mongo/mongo.go

```diff
@@ -105,10 +105,15 @@ func (m *Mongo) GetCacheTime(ctx context.Context, id string) (*models.CacheTime,
 // UpsertCacheTime adds or overrides an existing cache time
 func (m *Mongo) UpsertCacheTime(ctx context.Context, cacheTime *models.CacheTime) (err error) {
 	update := bson.M{
-		"$set": cacheTime,
+		"$set": bson.M{"path": cacheTime.Path, "etag": cacheTime.ETag, "collection_id": cacheTime.CollectionID, "release_time": cacheTime.ReleaseTime},
 	}
 	selector := bson.M{"_id": cacheTime.ID}
 
+	// Avoid storage of default releasetime zero value in the database.
+	if cacheTime.ReleaseTime != nil && cacheTime.ReleaseTime.IsZero() {
+		update["$unset"] = bson.M{"release_time": ""}
+	}
```

Comment:
Have you caught this behaviour in a test?

Reply:
This block has now been removed, it was left over from a previous commit. Testing for this scenario is covered in `upsert.feature` line at 65 onwards.